src/mac.c

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "provider.h"
+#include <string.h>
+
+struct p11prov_mac_ctx {
+    P11PROV_CTX *provctx;
+
+    P11PROV_OBJ *key;
+    P11PROV_SESSION *session;
+
+    CK_MECHANISM_TYPE mech;
+    CK_MECHANISM_TYPE hash_mech;
+};
+
+typedef struct p11prov_mac_ctx P11PROV_MAC_CTX;
+
+#define HMAC_MECHANISM(hash_mech) \
+    { \
+        .hash = CKM_##hash_mech, .mech = CKM_##hash_mech##_HMAC \
+    }
+
+static const struct {
+    CK_MECHANISM_TYPE mech;
+    CK_MECHANISM_TYPE hash;
+} hmac_mechanisms[] = { HMAC_MECHANISM(SHA_1),
+                        HMAC_MECHANISM(SHA224),
+                        HMAC_MECHANISM(SHA256),
+                        HMAC_MECHANISM(SHA384),
+                        HMAC_MECHANISM(SHA512),
+
+                        HMAC_MECHANISM(BLAKE2B_160),
+                        HMAC_MECHANISM(BLAKE2B_256),
+                        HMAC_MECHANISM(BLAKE2B_384),
+                        HMAC_MECHANISM(BLAKE2B_512),
+
+                        HMAC_MECHANISM(MD2),
```

Comment:
drop md2 and md5 as well as ripmed, we do not want to support obsolete and insecure hash functions really, and they are not needed for tls